middleware/keyauth/keyauth.go

```diff
@@ -19,23 +19,37 @@ const (
 	cookie = "cookie"
 )
 
+type extractorFunc func(c *fiber.Ctx) (string, error)
+
 // New creates a new middleware handler
 func New(config ...Config) fiber.Handler {
 	// Init config
 	cfg := configDefault(config...)
 
 	// Initialize
-	parts := strings.Split(cfg.KeyLookup, ":")
-	extractor := keyFromHeader(parts[1], cfg.AuthScheme)
-	switch parts[0] {
-	case query:
-		extractor = keyFromQuery(parts[1])
-	case form:
-		extractor = keyFromForm(parts[1])
-	case param:
-		extractor = keyFromParam(parts[1])
-	case cookie:
-		extractor = keyFromCookie(parts[1])
+
+	parts := strings.Split(cfg.KeyLookup, "|")
```

Comment:
I don't agree with this change, it's not standard practice. No other framework does something like this.

Reply:
@gaby - 

Do you mind elaborating slightly? Is your concern the entire concept of "multiple key lookup sources", or the way I tried to reuse the existing KeyLookup parameter? I'll admit it's a half-baked idea of mine rather than copying an example from Express - but I don't believe that they ship a standard keyauth middleware like this one to copy the API from.

If its the former, I'll just drop this PR entirely :)
If its the latter, I'll revise things to introduce a new parameter rather than overload the existing one. The only reason I didn't do this up front is that it implies we'll need to intelligently handle the case where a user provides _both_ parameters if I were to add say, a second parameter to Config named "KeyLookups" of type []string - in that case, I'm personally leaning to defaulting to the single "keylookup" parameter for backwards compatibility

Reply:
I realized the simple and best solution was to leave KeyLookup entirely alone, and add an "Additional" field. PR updated. Let me know if this addresses your concerns, @gaby

Reply:
Agree, much better 💪